include/bluetooth/services/hrs_client.h

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BT_HRS_CLIENT_H_
+#define BT_HRS_CLIENT_H_
+
+/**
+ * @file
+ * @defgroup bt_hrs_client Heart Rate Service Client
+ * @{
+ * @brief Heart Rate (HR) Service Client API.
+ */
+
+#include <bluetooth/conn.h>
+#include <bluetooth/gatt_dm.h>
+#include <bluetooth/gatt.h>
+
+#include <sys/atomic.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**@brief Heart Rate Service error codes
+ *
+ * This service defines the following Attribute Protocol Application error codes.
+ */
+enum bt_hrs_client_error {
+	/** Control Point value not supported. */
+	BT_HRS_CLIENT_ERROR_CP_NOT_SUPPORTED = 0x80
+};
+
+/**@brief Heart Rate Control Point values
+ *
+ * The values of the Control Point characteristic.
+ */
+enum bt_hrs_client_cp_value {
+	/** Resets the value if the Energy Expended field in
+	 * the Heart Rate Measurement characteristic to 0.
+	 */
+	BT_HRS_CLIENT_CP_VALUE_RESET_EE = 0x01
+};
+
+/**@brief Body sensor location values.
+ */
+enum bt_hrs_client_sensor_location {
+	/** Body Sensor Location: Other. */
+	BT_HRS_CLIENT_SENSOR_LOCATION_OTHER,
+
+	/** Body Sensor Location: Chest. */
+	BT_HRS_CLIENT_SENSOR_LOCATION_CHEST,
+
+	/** Body Sensor Location: Wrist. */
+	BT_HRS_CLIENT_SENSOR_LOCATION_WRIST,
+
+	/** Body Sensor Location: Finger. */
+	BT_HRS_CLIENT_SENSOR_LOCATION_FINGER,
+
+	/** Body Sensor Location: Hand. */
+	BT_HRS_CLIENT_SENSOR_LOCATION_HAND,
+
+	/** Body Sensor Location: Ear Lobe. */
+	BT_HRS_CLIENT_SENSOR_LOCATION_EAR_LOBE,
+
+	/** Body Sensor Location: Foot. */
+	BT_HRS_CLIENT_SENSOR_LOCATION_FOOT
+};
+
+/**@brief Heart Measurement flags structure.
+ */
+struct bt_hrs_flags {
+	/** Value Format flag. */
+	uint8_t value_format : 1;
+
+	/** Sensor Contact detected flag. */
+	uint8_t sensor_contact_detected : 1;
+
+	/** Sensor Contact supported flag. */
+	uint8_t sensor_contact_supported : 1;
+
+	/** Energy Expended preset flag. */
+	uint8_t energy_expended_present : 1;
+
+	/** RR-intervals present flag. */
+	uint8_t rr_intervals_present : 1;
+};
+
+/**@brief Data structure of the Heart Rate Measurement characteristic.
+ */
+struct bt_hrs_client_measurement {
+	/** Flags structure. */
+	struct bt_hrs_flags flags;
+
+	/** RR-intervals count. */
+	uint8_t rr_intervals_count;
+
+	/** RR-intervals represented by 1/1024 second as unit. Present if @ref rr_intervals_present
+	 * is set. The interval with index 0 is older then the interval with index 1.
+	 */
+	uint16_t rr_intervals[CONFIG_BT_HRS_CLIENT_RR_INTERVALS_COUNT];
+
+	/** Heart Rate Measurement Value in beats per minute unit. */
+	uint16_t hr_value;
+
+	/** Energy Expended in joule unit. Present if @ref energy_expended_present is set. */
+	uint16_t energy_expended;
+};
+
+struct bt_hrs_client;
```

Comment:
Please add a comment why this implicitly defined here.

Reply:
Ok, done